### PR TITLE
AddDialog: Show the supported language for each driver.

### LINKED ISSUE #156: Printer driver name repeated in list several times
### What Happened?

If you try to install a printer manually by selecting a manufacturer and then a driver, often there will be many duplicate entries in this list. See screenshot below: 
repeated_driver_name

I am guessing that although these names are duplicated, it might not be the same that all these entries refer to the same driver.

However, there is no way to tell them apart. Hovering over the driver name in the list or right-clicking it reveals no more information. So I am really unsure which to choose. I have to just pick one and hope it works, which is not user friendly.

### Steps to Reproduce

1. Open Settings and choose Printers.
2. Click + in the bottom left corner.
3. Choose a printer in the device list (it can be either local or network, it doesn't matter).
4. Choose a manufacturer in the left-hand column (e.g. Kyocera, but lots of manufacturers have repeated entries)
5. See lots of duplicates entries in the right-hand list.

### Expected Behavior

Each item in the right-hand list should either be unique, or there should be some way to find out more information about the driver so you can make an informed choice about which to install.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by tintou**
Fixes #156

### CODE REVIEW

**Comment by jeremypw** [comment]
This PR still results in (apparently) duplicate entries - all with the same language.  Afaict, this is due to faulty logic in the function `private void fetch_ppds (TempDevice temp_device) {` which can add the same driver multiple times.  I you wish I can push a PR to fix this but for now I will leave it to you.

**Review by jeremypw — CHANGES_REQUESTED** [review]
I think the attribute key for language is `attributes-natural-language` and this is not being handled when the driver is loaded.

**Comment by jeremypw** [comment]
I've tried adding most of the driver properties to the list and while this resolves some duplicates, there are some HP drivers that differ (afaict) only slightly by the ppd name (`hplip:1/ppd/...` `hplip:2/ppd/...` `hplip:3/ppd/...` etc).  In some cases (not HP) the product name is more informative.

Difficult to know what to do about this. One solution would be to only show recommended drivers by default (with an option to show all).  Also could add a facility to show all available properties for a given driver.  Could check whether the previous make and model was the same but different product name.

**Comment by jeremypw** [comment]
Example of near duplicate HP Drivers

HPPrinterDrivers

**Comment by jeremypw** [comment]
Need design input regarding how much information to show in the list (see also my comments of 4th Mar above).

**Comment by andrewdbate** [comment]
Just to add my two cents here, it seems like including the PPD name could be information overload but at the same time there has to be a way to uniquely identify each driver.

Would it be possible to only show the PPD name if it is necessary to make the entry unique?

For example, the above list shows multiple entries for `HP 2000c, hpcups 3.20.3 (en)`. In this case, it is necessary to include the PPD name for each because this is the only way to uniquely identify each entry. However, for `HP 2000C Foomatic/pcl3 (en)` the PPD name does not need to be included because the entry is unique without including it.

Not sure what your thoughts are on this suggestion design-wise.

Either way, it would be good if there was a way to get all information on a particular driver before adding the printer. For example, if there was another panel that shows all information, including the PPD name, for the currently selected driver in the list.

**Comment by danrabbit** [comment]
Showing everything on one line is indeed a lot and kind of hard to compare. Can we maybe split it up into two lines so you'd get something like:

**HP 2500c, hpcups 3.20.03 (recommended)**
hplip:1/ppd/hplip/HP/hp-2500c.ppd — en

**HP 2500c, hpcups 3.20.03**
hplip:2/ppd/hplip/HP/hp-2500c.ppd — en

**HP 2500c, hpcups 3.20.03**
drv:///hpcups.drv/hp-2500c.ppd — en

Goals here:
* Avoid cutting off information at smaller window widths
* Make it more likely that differentiating information lines up so that it's easier to compare
* Make sure "(recommended)" stands out and isn't buried in between other information
* Use `—` as a separator instead of `()` to more clearly differentiate that the lang code isn't information that is part of a driver name or location/uri/whatever that means. I put it at the end here because it looks like in Jeremy's screenshot that it's not very often differentiating for a row

**Comment by jeremypw** [comment]
Yes, a design which is necessary for HP printers may be overkill or otherwise sub-optimal for other manufacturers unfortunately but I guess that is inevitable.  Dani's suggestion is a reasonable compromise.

**Comment by danrabbit** [comment]
We could also switch around the two parts of the bottom line if that's more important. I think that's still not too hard to compare:

**HP 2500c, hpcups 3.20.03 (recommended)**
en — hplip:1/ppd/hplip/HP/hp-2500c.ppd

**HP 2500c, hpcups 3.20.03**
fr — hplip:2/ppd/hplip/HP/hp-2500c.ppd

**HP 2500c, hpcups 3.20.03**
es — drv:///hpcups.drv/hp-2500c.ppd

File: src/Dialogs/AddDialog.vala [hunk 1]
```diff
@@ -263,18 +262,9 @@ public class Printers.AddDialog : Granite.Dialog {
...
-        driver_view.headers_visible = false;
-        driver_view.get_selection ().mode = Gtk.SelectionMode.BROWSE;
-        driver_view.set_tooltip_column (0);
-        driver_view.set_search_column (0);
-
-        var driver_cellrenderer = new Gtk.CellRendererText ();
-        driver_cellrenderer.ellipsize_set = true;
-        driver_cellrenderer.ellipsize = Pango.EllipsizeMode.END;
-        driver_view.insert_column_with_attributes (-1, null, driver_cellrenderer, "text", 0);
+        // driver_list_store = new ListStore (typeof (DeviceDriver));
```
Comment (danrabbit) [inline src/Dialogs/AddDialog.vala:265]:
```suggestion
```

**Review by danrabbit — APPROVED** [review]
Iterated on the design in the other branch, so this looks good to me

**Review by jeremypw — APPROVED** [review]
LGTM

**Pull request merged**